### PR TITLE
Charlesmchen/auto size contact cells

### PR DESCRIPTION
**Pull request by charlesmchen-signal**
The only place I haven't fully converted is `ContactPicker`.

PTAL @michaelkirk

### CODE REVIEW

File: Signal/src/ViewControllers/AppSettings/BlockListViewController.m [hunk 1]
```diff
@@ -78,18 +79,22 @@ - (void)updateTableContents
...
-            if (signalAccount) {
-                [cell configureWithSignalAccount:signalAccount contactsManager:helper.contactsManager];
-            } else {
-                [cell configureWithRecipientId:phoneNumber contactsManager:helper.contactsManager];
-            }
-
-            return cell;
-        }
-                                      customRowHeight:[ContactTableViewCell rowHeight]
+        [blocklistSection addItem:[OWSTableItem
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/AppSettings/BlockListViewController.m]: Is your clang format feeling OK?

I never really know what to expect, so this could be correct... just wanted to double check.

File: Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m [hunk 2]
```diff
@@ -44,7 +44,6 @@ - (void)viewDidLoad
 
     self.isExpectingMoreDevices = NO;
     self.tableView.rowHeight = UITableViewAutomaticDimension;
-    self.tableView.estimatedRowHeight = 70;
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: Why are you removing the estimatedRowHeights here and elsewhere?

File: SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m [hunk 3]
```diff
@@ -450,7 +450,7 @@ - (void)ensureThumbnail
     }
 
     if (result == nil) {
-        OWSFail(@"%@ Unable to build thumnail for attachmentId: %@", self.logTag, self.uniqueId);
```
Comment (michaelkirk-signal) [inline SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m]: Why this change? Any clue what got you into this state?

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m]: I've been testing how we handle strange media formats. There are definitely some container/codec combinations where iOS can't extract a thumbnail.  I don't think we'll be able to detect these beforehand (and treat them as a generic attachment).

File: Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m (hunk 2 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: We could go either way here.

`estimatedRowHeight` improves perf for tables with many cells.  But I'm not sure that we're supplying a good estimate since our cell heights can vary wildly based on dynamic text size.  The docs don't make clear to me if an inaccurate estimate is better or worse for perf than no estimate at all.  WDYT?

File: Signal/src/ViewControllers/AppSettings/BlockListViewController.m (hunk 1 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/AppSettings/BlockListViewController.m]: I have no idea how to determine that.  It's a mysterious beast.

**Comment by charlesmchen-signal** [comment]
PTAL @michaelkirk

File: Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m (hunk 2 above)
Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: I would leave it as is. It's true we don't know the inner workings of estimatedRowHeight, but based on the documentation, we should be providing a reasonable guess, which we were.

File: SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m (hunk 3 above)
Comment (michaelkirk-signal) [reply SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m]: I've hit this assert a few times and it was always a result of some error in our attachment processing logic, so I would have kept it as a good sentinel. 

> I don't think we'll be able to detect these beforehand

We don't have to detect them all - this is just a debug assert. But if you think we'll be hitting this branch under normal usage in such a way that it's more distracting than the sentinel value it provides, I guess we can remove it.

File: Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m (hunk 2 above)
Comment (charlesmchen-signal) [reply Signal/src/ViewControllers/AppSettings/OWSLinkedDevicesTableViewController.m]: I've already put it back in another PR.  You were right!

File: SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m (hunk 3 above)
Comment (charlesmchen-signal) [reply SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m]: Okay, I'll put it back.
